### PR TITLE
Add <spotify> tag

### PR DESCRIPTION
**Pull request by sirKiraUzumaki**
Adding a <spotify> tag to embed song from Spotify to ps.

Example: 
umtrBapwb6ll

Just like <youtube> tag, it will play songs on ps.

### CODE REVIEW

File: src/battle-log.ts [hunk 1]
```diff
@@ -805,7 +805,16 @@ class BattleLog {
...
-			} else if (tagName === 'youtube') {
+			} else if (tagName === 'spotify') {
+				// <iframe src="https://open.spotify.com/embed/track/6aSYnCIwcLpnDXngGKAEzZ" width="300" height="380" frameborder="0" allowtransparency="true" allow="encrypted-media"></iframe>
+				const src = getAttrib('src');
+				const songId = /(?:\?v=|\/track\/)([A-Za-z0-9]+)/.exec(src)[1];
+						
+            return {
+				   tagName: 'iframe',
+					attribs: ['src', `https://open.spotify.com/embed/track/${songId}`,'width', '300', 'height', '380', 'frameborder', '0', 'allowtransparency', 'true', 'allow', 'encrypted-media'],
+				};
```
Comment (KrisXV) [inline src/battle-log.ts:816]:
```suggestion
				return {
					tagName: 'iframe',
					attribs: ['src', `https://open.spotify.com/embed/track/${songId}`,'width', '300', 'height', '380', 'frameborder', '0', 'allowtransparency', 'true', 'allow', 'encrypted-media'],
				};
```

File: src/battle-log.ts [hunk 2]
```diff
@@ -805,7 +805,16 @@ class BattleLog {
 				const color = this.usernameColor(toID(getAttrib('name')));
 				const style = getAttrib('style');
 				setAttrib('style', `${style};color:${color}`);
-			} else if (tagName === 'youtube') {
+			} else if (tagName === 'spotify') {
+				// <iframe src="https://open.spotify.com/embed/track/6aSYnCIwcLpnDXngGKAEzZ" width="300" height="380" frameborder="0" allowtransparency="true" allow="encrypted-media"></iframe>
+				const src = getAttrib('src');
+				const songId = /(?:\?v=|\/track\/)([A-Za-z0-9]+)/.exec(src)[1];
```
Comment (KrisXV) [inline src/battle-log.ts:811]:
```suggestion
				const songId = /(?:\?v=|\/track\/)([A-Za-z0-9]+)/.exec(src)?.[1];
```

File: src/battle-log.ts [hunk 3]
```diff
@@ -805,7 +805,16 @@ class BattleLog {
...
+						
+            return {
+				   tagName: 'iframe',
+					attribs: ['src', `https://open.spotify.com/embed/track/${songId}`,'width', '300', 'height', '380', 'frameborder', '0', 'allowtransparency', 'true', 'allow', 'encrypted-media'],
+				};
+		   } else if (tagName === 'youtube') {
```
Comment (Zarel) [inline src/battle-log.ts:817]: The indentation on this line is wrong, too, which is what's leading to it being part of the diff:

```suggestion
			} else if (tagName === 'youtube') {
```

File: src/battle-log.ts (hunk 2 above)
Comment (AnnikaCodes) [reply src/battle-log.ts:811]: There should probably be a check to not return a tag if `songId` is undefined. I'm not familar with the tag parser but https://open.spotify.com/embed/track/undefined doesn't seem like a very valid URL :p

File: src/battle-log.ts [hunk 4]
```diff
@@ -805,6 +805,15 @@ class BattleLog {
 				const color = this.usernameColor(toID(getAttrib('name')));
 				const style = getAttrib('style');
 				setAttrib('style', `${style};color:${color}`);
+			} else if (tagName === 'spotify') {
+				// <iframe src="https://open.spotify.com/embed/track/6aSYnCIwcLpnDXngGKAEzZ" width="300" height="380" frameborder="0" allowtransparency="true" allow="encrypted-media"></iframe>
+				const src = getAttrib('src');
```
Comment (sirKiraUzumaki) [inline src/battle-log.ts:810]:
```suggestion
				const src = getAttrib('src') || '';
```

**Comment by Zarel** [comment]
Looks good; thanks!

**Pull request merged**